common/src/main/java/com/wynntils/screens/settings/WynntilsSettingsScreen.java

```diff
@@ -0,0 +1,239 @@
+                (int) (BAR_HEIGHT + 25),
+                (int) (this.width / 6f),
+                SEARCH_BAR_HEIGHT,
+                newSearchText -> featureList.reAddEntriesWithSearchFilter(newSearchText),
+                this);
+
+        this.featureSettingWidget = new FeatureSettingWidget(
+                this.width / 5, (int) BAR_HEIGHT, this.width / 5 * 4, (int) (this.height - BAR_HEIGHT * 2), this);
+
+        this.addRenderableWidget(this.searchWidget);
+        this.addRenderableWidget(this.featureSettingWidget);
+
+        this.setFocused(searchWidget);
+    }
+
+    @Override
+    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        this.renderBackground(poseStack);
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        I18n.get("screens.wynntils.settingsScreen.title"),
+                        this.width / 2f,
+                        10,
+                        CommonColors.WHITE,
+                        FontRenderer.TextAlignment.CENTER_ALIGNED,
+                        FontRenderer.TextShadow.OUTLINE);
+
+        featureList.render(poseStack, mouseX, mouseY, partialTick);
+
+        // Re-render the bottom bar again, so overlaying buttons are not displayed
+        RenderUtils.drawRect(poseStack, FOREGROUND_COLOR, 0, this.height - BAR_HEIGHT, 10, this.width, BAR_HEIGHT);
+
+        poseStack.pushPose();
+
+        poseStack.translate(this.width / 90f, 0, 0);
+
+        FontRenderer.getInstance()
+                .renderAlignedTextInBox(
+                        poseStack,
+                        I18n.get("screens.wynntils.settingsScreen.featureList"),
+                        0,
+                        this.width / 6f,
+                        BAR_HEIGHT + 7.5f,
+                        0,
+                        CommonColors.WHITE,
+                        FontRenderer.TextAlignment.CENTER_ALIGNED,
+                        FontRenderer.TextShadow.OUTLINE);
+
+        poseStack.popPose();
+
+        super.render(poseStack, mouseX, mouseY, partialTick);
+    }
+
+    @Override
+    public void renderBackground(PoseStack poseStack) {
+        RenderUtils.drawRect(poseStack, BACKGROUND_COLOR, 0, 0, 0, this.width, this.height);
+
+        RenderUtils.drawRect(poseStack, FOREGROUND_COLOR, 0, 0, 0, this.width, BAR_HEIGHT);
+        RenderUtils.drawRect(poseStack, FOREGROUND_COLOR, 0, this.height - BAR_HEIGHT, 0, this.width, BAR_HEIGHT);
+    }
+
+    @Override
+    public void resize(Minecraft minecraft, int width, int height) {
+        this.init();
+        super.resize(minecraft, width, height);
+    }
+
+    @Override
+    public boolean mouseScrolled(double mouseX, double mouseY, double delta) {
+        if (featureList.isMouseOver(mouseX, mouseY)) {
+            return featureList.mouseScrolled(mouseX, mouseY, delta);
+        } else if (featureSettingWidget.isMouseOver(mouseX, mouseY)) {
```

Comment:
Is this not handled by super, since `featureSettingWidget` is set by `addRenderableWidget`?

Reply:
This is needed so the "order" of processing the click is how I want it, compared to super.

Reply:
I might be wrong here, but if not, that would also make a case for also adding the `featureList` to the renderable widget list. I think that is in general more akin to how the guis of minecraft work. I understand this has been in the works for long, and I will not require any additional works just "because it looks better", but if you agree with me you might consider this :)

Reply:
I don't think feature list can be added as a widget though.

Reply:
(because of the superclass)

Reply:
ok